From bce132a8578455f6fb4a618c4b5332cd2ae3a455 Mon Sep 17 00:00:00 2001
From: LongLuu <long.luu.ur@renesas.com>
Date: Fri, 5 Feb 2021 17:42:47 +0700
Subject: [PATCH 024/135] tty: serial: sh-sci: Support release reset state for
 SCIF

This commit support release reset state for SCIF module

Signed-off-by: LongLuu <long.luu.ur@renesas.com>
---
 drivers/tty/serial/sh-sci.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index d89e701..b367a26 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -51,6 +51,7 @@
 #include <linux/timer.h>
 #include <linux/tty.h>
 #include <linux/tty_flip.h>
+#include <linux/reset.h>
 
 #ifdef CONFIG_SUPERH
 #include <asm/sh_bios.h>
@@ -125,6 +126,7 @@ struct sci_port {
 	unsigned int		sampling_rate_mask;
 	resource_size_t		reg_size;
 	struct mctrl_gpios	*gpios;
+	struct reset_control *rstc;
 
 	/* Clocks */
 	struct clk		*clks[SCI_NUM_CLKS];
@@ -2903,6 +2905,13 @@ static int sci_init_single(struct platform_device *dev,
 	port->mapbase = res->start;
 	sci_port->reg_size = resource_size(res);
 
+	sci_port->rstc = devm_reset_control_get(&dev->dev, NULL);
+
+	if (IS_ERR(sci_port->rstc))
+		dev_warn(&dev->dev, "Failed to get reset controler\n");
+	else
+		reset_control_deassert(sci_port->rstc);
+
 	for (i = 0; i < ARRAY_SIZE(sci_port->irqs); ++i)
 		sci_port->irqs[i] = platform_get_irq(dev, i);
 
-- 
2.7.4

